src/group_theory/schur_zassenhaus.lean

```diff
@@ -162,21 +162,206 @@ begin
+  exact ⟨⟨λ g h, eq_top_iff.mp ((step3 h1 h2 h3 N.center).resolve_left
+    (step6 h1 h2 h3).bot_lt_center.ne') (mem_top h) g⟩⟩,
+end
+
+/-- Do not use this lemma: It is made made obsolete by `exists_right_complement'_of_coprime` -/
+lemma step8 : false :=
+begin
+  haveI := step7 h1 h2 h3,
+  exact not_exists_of_forall_not h3 (exists_right_complement'_of_coprime_aux h1),
+end
+
+end schur_zassenhaus_induction
+
+variables {n : ℕ} {G : Type u} [group G]
+
+/-- Do not use this lemma: It is made made obsolete by `exists_right_complement'_of_coprime` -/
+lemma exists_right_complement'_of_coprime_aux' [fintype G] (hG : fintype.card G = n)
+  {N : subgroup G} [N.normal] (hN : nat.coprime (fintype.card N) N.index) :
+  ∃ H : subgroup G, is_complement' N H :=
+begin
+  tactic.unfreeze_local_instances,
+  revert G,
+  apply nat.strong_induction_on n,
+  rintros n ih G _ _ rfl N _ hN,
+  exact not_forall_not.mp
+    (schur_zassenhaus_induction.step8 hN (λ G' _ _ hG', by { apply ih _ hG', refl })),
+end
+
+/-- **Schur-Zassenhaus** for normal subgroups:
+  If `H : subgroup G` is normal, and has order coprime to its index, then there exists a
+  subgroup `K` which is a (right) complement of `H`. -/
+theorem exists_right_complement'_of_coprime [fintype G]
+  {N : subgroup G} [N.normal] (hN : nat.coprime (fintype.card N) N.index) :
+  ∃ H : subgroup G, is_complement' N H :=
+exists_right_complement'_of_coprime_aux' rfl hN
+
+/-- **Schur-Zassenhaus** for normal subgroups:
+  If `H : subgroup G` is normal, and has order coprime to its index, then there exists a
+  subgroup `K` which is a (right) complement of `H`. -/
+theorem exists_right_complement'_of_coprime'
+  {N : subgroup G} [N.normal] (hN : nat.coprime (nat.card N) N.index) :
+  ∃ H : subgroup G, is_complement' N H :=
+begin
+  by_cases hN1 : nat.card N = 0,
+  { rw [hN1, nat.coprime_zero_left, index_eq_one] at hN,
+    rw hN,
+    exact ⟨⊥, is_complement'_top_bot⟩ },
+  by_cases hN2 : N.index = 0,
+  { rw [hN2, nat.coprime_zero_right] at hN,
+    haveI := (cardinal.to_nat_eq_one_iff_unique.mp hN).1,
+    rw N.eq_bot_of_subsingleton,
+    exact ⟨⊤, is_complement'_bot_top⟩ },
+  have hN3 : nat.card G ≠ 0,
+  { rw ← N.card_mul_index,
+    exact mul_ne_zero hN1 hN2 },
+  haveI := (cardinal.lt_omega_iff_fintype.mp
+    (lt_of_not_ge (mt cardinal.to_nat_apply_of_omega_le hN3))).some,
+  rw nat.card_eq_fintype_card at hN,
+  exact exists_right_complement'_of_coprime hN,
+end
+
+/-- **Schur-Zassenhaus** for normal subgroups:
+  If `H : subgroup G` is normal, and has order coprime to its index, then there exists a
+  subgroup `K` which is a (left) complement of `H`. -/
+theorem exists_left_complement'_of_coprime
+  [fintype G] {N : subgroup G} [N.normal] (hN : nat.coprime (fintype.card N) N.index) :
+  ∃ H : subgroup G, is_complement' H N :=
+Exists.imp (λ _, is_complement'.symm) (exists_right_complement'_of_coprime hN)
+
+/-- **Schur-Zassenhaus** for normal subgroups:
```

Comment:
is this thm duplicated?

Reply:
It's slightly different (no fintype assumption)

Reply:
It's slightly different (e.g., no fintype assumption).

Reply:
sorry, I left a comment on a review and forgot to press send. my point is at the moment it seems like `lemma refl {α : Type*} [fintype α] : α = α` and also `lemma refl' {α : Type*} : α = α`. Why not just have the non-fintype version and make the fintype version a private lemma too?

Reply:
They are phrased slightly differently. The fintype version uses `fintype.card`, whereas the non-fintype version uses `nat.card`.

Reply:
I think it's nice to have both, so that if you're working with `fintype.card`, you don't have to manually convert to `nat.card` in order to apply the theorem.

Reply:
ohh, I am blind. Yes, that seems reasonable!